templates/custom/projects-cm.yaml

```diff
@@ -0,0 +1,91 @@
+{{- $root := . -}}
+{{- range $project := .Values.projects.project }}
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ $project.name}}-project-cm
+data:
+  projects.sh: |
+   #!/usr/bin/env bash
+
+    # exit when any command fails
+    set -e
+
+    coreHostUrl={{ $root.Values.expose.ingress.hosts.core }}
+    basicAuthCreds="${HARBOR_ADMIN_USERNAME}:${HARBOR_ADMIN_PASSWORD}"
+
+    function isProjectExists(){
+      # call project API and return the value.
+      res=$(curl -I HEAD "$coreHostUrl/api/projects?project_name=$1" \
+      -H "accept: application/json" \
+      -u $basicAuthCreds)
+      echo "response: " $res
+    }
+
+    function createProject() {
+      # call Project API and create the project with the values taken from the values yaml
+      project=$(curl -X POST "$coreHostUrl/api/projects" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_ADMIN_USERNAME}":"${HARBOR_PASSWORD}" \
+      -d "{ \"project_name\": \"$1\"}")
+      echo "Project created : ${project}"
+      echo ${project}
+    }
+
+    function updateProjectWithMemberGroup() {
+      projectName=$1
+      name=$(IFS=$',' && echo $1 | awk -F '-' '{print $1}')
+      role_id=$(IFS=$',' && echo $1 | awk -F '-' '{print $2}')
+
+      # Get projecid for the given projectname.
+      projectID=$(curl -X GET "$coreHostUrl/api/projects?name=$projectName" \
+      -H "accept: application/json" \
+      -H "Content-Type: application/json" \
+      -u "${HARBOR_USER_NAME}":"${HARBOR_ADMIN_PASSWORD}"  | jq '.[].project_id ')
```

Comment:
replace with `basicAuthCreds`

Reply:
Resolved and use variable 'basicAuthCreds'